src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -174,19 +166,21 @@ export class SavedObjectsRepository {
     }
 
     const allowedTypes = [...new Set(visibleTypes.concat(includedHiddenTypes))];
-
-    return new injectedConstructor({
-      index: indexName,
-      migrator,
-      mappings,
-      typeRegistry,
-      serializer,
-      allowedTypes,
-      client,
-    });
+    return new injectedConstructor(
+      {
+        index: indexName,
+        migrator,
+        mappings,
+        typeRegistry,
+        serializer,
+        allowedTypes,
+        client,
+      },
+      postgresClient
```

Comment:
Should `postgresClient` be a optional parameter for ` SavedObjectsRepository` or provide a more generic parameter name? As for `OpensearchSavedObjectsRepository` implementation of ` SavedObjectsRepository`, `postgresClient` parameter is not required. In the future if we add another implementation of the ` SavedObjectsRepository`, how are we going to handle db client for a new db type?